openlibrary/plugins/openlibrary/js/realtime_account_validation.js

```diff
@@ -0,0 +1,99 @@
+import { ugettext } from './i18n';
+
+export function initRealTimeValidation() {
+
+    $('#username').on('keyup', function(){
+        var value = $(this).val();
+        $('#userUrl').addClass('darkgreen').text(value).css('font-weight','700');
+    });
+
+    function validateUsername() {
+        var value = $(this).val();
+        if (!value=='') {
+            $.ajax({
+                url: `/account/validate?username=${value}`,
+                type: 'GET',
+                success: function(errors) {
+                    if (errors.username) {
+                        $(document.getElementById('username')).removeClass().addClass('required invalid');
+                        $('label[for="username"]').removeClass().addClass('invalid');
+                        $('#usernameMessage').removeClass().addClass('invalid').html(`${errors.username}<br>`);
+                    } else {
+                        $('#usernameMessage').removeClass().addClass('darkgreen').html('<br>');
+                        $('label[for="username"]').removeClass();
+                        $(document.getElementById('username')).removeClass().addClass('required');
+                    }
+                }
+            });
+        }
+        else {
+            $('label[for="username"]').removeClass();
+            $(document.getElementById('username')).removeClass().addClass('required');
+            $('#usernameMessage').removeClass().html('<br>');
+        }
+    }
+
+    function validateEmail() {
+        var value_email = $(this).val();
+        if (!value_email=='') {
+            $.ajax({
+                url: `/account/validate?email=${value_email}`,
```

Comment:
We may have to urlencode `+` for the email. Otherwise, the API call will treat it as a space. This would means we would also have to urlencode the email on the other side.

`encodeURIComponent` from js (before we do the ajax request)
`unquote_plus` on the backend (within API)